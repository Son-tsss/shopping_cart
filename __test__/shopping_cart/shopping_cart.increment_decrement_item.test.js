import ShoppingCart from '../../src/shopping_cart'

it('should increment existing item in the items list', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.incrementItem(newItem.id)

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('incrementing not existing item should change nothing', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.incrementItem("i3")

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('should decrement existing item in the items list ', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.decrementItem(newItem.id)

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('decrementing not existing item should change nothing ', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.decrementItem("i3")

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

it('shouldn`t decrement item below 0', ()=>{
    const shoppingCart = new ShoppingCart

    const newItem  = {
        id: "i1",
        price: 100
    }

    shoppingCart.addItem(newItem)
    shoppingCart.decrementItem(newItem.id)
    shoppingCart.decrementItem(newItem.id)

    const real = shoppingCart.historyState

    expect(real).toMatchSnapshot()
})

