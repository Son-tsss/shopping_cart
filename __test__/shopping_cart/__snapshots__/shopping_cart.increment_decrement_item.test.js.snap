// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrementing not existing item should change nothing  1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 1,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;

exports[`incrementing not existing item should change nothing 1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 1,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;

exports[`should decrement existing item in the items list  1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 0,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;

exports[`should increment existing item in the items list 1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 2,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;

exports[`shouldn\`t decrement item below 0 1`] = `
Object {
  "currentState": Object {
    "discount": 1,
    "items": Array [
      Object {
        "count": 0,
        "discount": 1,
        "id": "i1",
        "price": 100,
      },
    ],
  },
  "futureStates": Array [],
  "previousStates": Array [
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 1,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
    Object {
      "discount": 1,
      "items": Array [
        Object {
          "count": 0,
          "discount": 1,
          "id": "i1",
          "price": 100,
        },
      ],
    },
  ],
}
`;
